commands/FBPrintCommands.py

```diff
@@ -424,3 +425,60 @@ def run(self, arguments, options):
+  def name(self):
+    return 'pcurl'
+
+  def description(self):
+    return 'Print the NSURLRequest (HTTP) as curl command.'
+
+  def options(self):
+    return [
+      fb.FBCommandArgument(short='-e', long='--embed-data', arg='embed', boolean=True, default=False, help='Embed request data as base64.'),
+    ]
+
+  def args(self):
+    return [ fb.FBCommandArgument(arg='request', type='NSURLRequest*/NSMutableURLRequest*', help='The request to convert to the curl command.') ]
+
+  def run(self, arguments, options):
+    request = arguments[0]
+    HTTPHeaderSring = ''
+    HTTPMethod = fb.evaluateExpressionValue('(id)[{} HTTPMethod]'.format(request)).GetObjectDescription()
+    URL = fb.evaluateExpressionValue('(id)[{} URL]'.format(request)).GetObjectDescription()
+    timeout = fb.evaluateExpression('(NSTimeInterval)[{} timeoutInterval]'.format(request))
+    HTTPHeaders = fb.evaluateObjectExpression('(id)[{} allHTTPHeaderFields]'.format(request))
+    HTTPHeadersCount = fb.evaluateIntegerExpression('[{} count]'.format(HTTPHeaders)) 
+    allHTTPKeys = fb.evaluateObjectExpression('[{} allKeys]'.format(HTTPHeaders))
+    for index in range(0, HTTPHeadersCount):
+        key = fb.evaluateObjectExpression('[{} objectAtIndex:{}]'.format(allHTTPKeys, index))
+        keyDescription = fb.evaluateExpressionValue('(id){}'.format(key)).GetObjectDescription()
+        value = fb.evaluateExpressionValue('(id)[(id){} objectForKey:{}]'.format(HTTPHeaders, key)).GetObjectDescription()
+        if len(HTTPHeaderSring) > 0:
+            HTTPHeaderSring += ' '
+        HTTPHeaderSring += '-H "{}: {}"'.format(keyDescription, value)
+    HTTPData = fb.evaluateObjectExpression('[{} HTTPBody]'.format(request))
+    dataFile = None
+    dataAsString = None
+    if fb.evaluateIntegerExpression('[{} length]'.format(HTTPData)) > 0:
+        dataFile = '/tmp/curl_data_{}'.format(fb.evaluateExpression('(NSTimeInterval)[NSDate timeIntervalSinceReferenceDate]'))
+        if options.embed:
+          if fb.evaluateIntegerExpression('[{} respondsToSelector:@selector(base64EncodedStringWithOptions:)]'.format(HTTPData)):
+            dataAsString = fb.evaluateExpressionValue('(id)[(id){} base64EncodedStringWithOptions:0]'.format(HTTPData)).GetObjectDescription()
+        elif not runtimeHelpers.isIOSDevice():
+          fb.evaluateExpression('(BOOL)[{} writeToFile:@"{}" atomically:NO]'.format(HTTPData, dataFile))
+        else:
+          print 'HTTPBody data for iOS Device is supported only with "--portable" flag'
+          return False
+
+    commandString = ''
+    if dataAsString is not None and len(dataAsString) > 0:
+      commandString += 'echo "{}" | base64 -D -o "{}"; '.format(dataAsString, dataFile)
+    commandString += 'curl -X {} --connect-timeout {}'.format(HTTPMethod, timeout)
+    if len(HTTPHeaderSring) > 0:
+        commandString += ' ' + HTTPHeaderSring
+    if dataFile is not None:
+        commandString += ' --data-binary @"{}"'.format(dataFile)
```

Comment:
@mattjgalloway actually same file are used in two cases:
1. To embed data
2. To write data to file

If I will place name generation under the `not runtimeHelpers.isIOSDevice():` condition, then it will be not available here with `--embed-data` option.